src/index.js

```diff
@@ -206,7 +207,39 @@ function sanitizeHtml(html, options, _recursing) {
                 return;
               }
             }
-
+            if (name === 'iframe' && a === 'src') {
+              try {
+                var parsed = url.parse(value)
+                var whitelistedDomains = options.allowedIframeDomains.find(function(domain) {
+                  return domain === parsed.hostname;
+                })
+                if (!whitelistedDomains) {
+                  //check the href to see if the hostname didn't get picked up, doesn't mean it's invalid
+                  var noHost = options.allowedIframeDomains.find(function(domain) {
+                    return parsed.href.includes(domain)
```

Comment:
This looks like it would match the domain name anywhere in the URL, which is insecure as discussed.

Reply:
OK, I see how the check further down addresses this, but this entire branch of the code can be eliminated by fixing the // URLs and letting them go through url.parse normally.

Reply:
Yeah took this part out.